osu.Game/Screens/Select/Carousel/CarouselItem.cs

```diff
@@ -43,15 +41,20 @@ protected CarouselItem()
         /// <summary>
         /// Create a fresh drawable version of this item.
         /// </summary>
-        public abstract DrawableCarouselItem CreateDrawableRepresentation();
+        public abstract DrawableCarouselItem? CreateDrawableRepresentation();
 
         public virtual void Filter(FilterCriteria criteria)
         {
         }
 
         public virtual int CompareTo(FilterCriteria criteria, CarouselItem other) => ItemID.CompareTo(other.ItemID);
 
-        public int CompareTo(CarouselItem other) => CarouselYPosition.CompareTo(other.CarouselYPosition);
+        public int CompareTo(CarouselItem? other)
+        {
+            if (other == null) return 1;
```

Comment:
Refer to [docs](https://learn.microsoft.com/en-us/dotnet/api/system.icomparable-1.compareto?redirectedfrom=MSDN&view=net-7.0#remarks):

> By definition, any object compares greater than `null`, and two null references compare equal to each other.

There's no need to handle the latter case here as `this` cannot be `null` at the point of calling this method.

Also see [example](https://learn.microsoft.com/en-us/dotnet/api/system.icomparable-1.compareto?redirectedfrom=MSDN&view=net-7.0#examples).

Reply:
will keep in mind moving forward, and as usual thanks